src/dao/hooks/useDAOProposals.tsx

```diff
@@ -81,6 +89,28 @@ export function useDAOProposals(
     }
   }
 
+  async function getUpgradeProposalData(proposalData: DAOUpgradeProposal) {
```

Comment:
Move this function to the service layer (DAO service). This is essentially an API endpoint used for fetching Contract Upgrade Proposal data.